### PR TITLE
Complete `getindex` tests.

### PR DESCRIPTION
**Pull request by codetalker7**
This PR is to complete some `getindex` tests to increase coverage.

### CODE REVIEW

File: test/getindex.jl [hunk 1]
```diff
@@ -397,3 +401,24 @@ test_types(ts[ind])
                      Hour(-1), Minute(0), Second(0), Millisecond(1001))) == []
 @test index(getindex(tsdatetimes, Year(2007), Month(1), Day(2),
                      Hour(0), Minute(0), Second(0), Millisecond(0))) == [DateTime(2007, 01, 02, 0, 0, 0)]
+
+# getindex(ts::TS, y::Year, q::Quarter)
+y = Year(2007); q = Quarter(1)
```
Comment (chiraganand) [inline test/getindex.jl:406]: Can you test with `Quarter(2)` and `Quarter(5)`? The `1` case often gets passed in almost all kinds of tests.

File: test/getindex.jl [hunk 2]
```diff
@@ -397,3 +401,24 @@ test_types(ts[ind])
...
+dates = Date(2007, 1, 1):Day(1):Date(2007, 3, 31)
+t = ts[y, q]
+test_types(t)
+@test t[:, :Index] == dates
+@test t[:, :data] == data_vector[1:length(dates)]
+
+# getindex(ts::TS, y::Year, m::Month, d::Day)
+y = Year(2007); m = Month(1); d = Day(1)
```
Comment (chiraganand) [inline test/getindex.jl:414]: Similarly, let us test with `Month(2)` and `Month(13)` to cover some boundary conditions.

**Review by chiraganand — CHANGES_REQUESTED** [review]

**Review by chiraganand — APPROVED** [review]
